model-optimizer/extensions/middle/layer_normalization.py

```diff
@@ -0,0 +1,51 @@
+# Copyright (C) 2018-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+
+from extensions.ops.elementwise import Add, Mul
+from extensions.ops.mvn import MVN
+from mo.front.common.partial_infer.utils import int64_array
+from mo.front.tf.graph_utils import create_op_node_with_second_input
+from mo.graph.graph import Graph
+from mo.middle.replacement import MiddleReplacementPattern
+from mo.ops.squeeze import Squeeze
+from mo.ops.unsqueeze import Unsqueeze
+
+
+class LayerNormalization(MiddleReplacementPattern):
+    """
+    Decompose LayerNormalization to scale * (Squeeze - MVN(x) - Unsqueeze) + B
+    """
+    enabled = True
+
+    def find_and_replace_pattern(self, graph: Graph):
+        for node in graph.get_op_nodes(op='LayerNorm'):
+            axis = 1
+            if node.has('axis'):
+                axis = node.soft_get('axis')
+
+            input_rank = len(node.in_port(0).data.get_shape())
+            if axis < 0:
+                axis = input_rank + axis
```

Comment:
Just double check. Shouldn't we use input_rank -1 here because axis used in MVN after Squeeze?